dsl_parser/tests/test_deafult_import_resolver.py

```diff
@@ -155,35 +180,66 @@ def _test_default_resolver(self, import_url, rules,
+                        'Timeout while trying to import')
+                elif url == BAD_RESPONSE_CODE_URL:
+                    self.status_code = 404
+                    self.text = 404
+                elif url == RETRY_URL:
+                    if len(number_of_attempts) < DEFAULT_NUMBER_RETRIES:
+                        raise requests.ConnectionError(
+                            'Timeout while trying to import')
+                    else:
+                        return None
 
-        def mock_urlopen(url):
-            urls_to_resolve.append(url)
-            if url in [ORIGINAL_V1_URL, ORIGINAL_V2_URL, INVALID_V1_URL]:
-                raise urllib2.URLError('invalid url: {0}'.format(url))
-            elif url == ILLEGAL_URL:
-                raise ValueError('unknown url type: {0}'.format(url))
-            elif url in [VALID_V1_URL, VALID_V2_URL]:
-                return mock.MagicMock()
-
-        resolver = DefaultImportResolver(rules)
-        with mock.patch('urllib2.urlopen', new=mock_urlopen):
-            try:
-                resolver.resolve(import_url=import_url)
-                if expected_failure:
-                    err_msg = 'resolve should have been failed'
+        resolver = DefaultImportResolver(rules=rules)
+        with mock.patch('requests.get', new=mock_requests_get,
+                        create=True):
+            with mock.patch(
+                    'dsl_parser.import_resolver.abstract_import_resolver.'
+                    'DEFAULT_REQUEST_TIMEOUT', new=REQUEST_TIMEOUT):
+                try:
+                    resolver.resolve(import_url=import_url)
+                    if expected_failure:
+                        err_msg = 'resolve should have been failed'
+                        if partial_err_msg:
+                            err_msg = \
+                                '{0} with error message that contains: {1}'\
+                                .format(err_msg, partial_err_msg)
+                        raise AssertionError(err_msg)
+                except DSLParsingLogicException, ex:
+                    if not expected_failure:
+                        raise ex
                     if partial_err_msg:
-                        err_msg = '{0} with error message that contains: {1}'\
-                            .format(err_msg, partial_err_msg)
-                    raise AssertionError(err_msg)
-            except DSLParsingLogicException, ex:
-                if not expected_failure:
-                    raise ex
-                if partial_err_msg:
-                    self.assertIn(partial_err_msg, str(ex))
+                        self.assertIn(partial_err_msg, str(ex))
 
         self.assertEqual(len(expected_urls_to_resolve), len(urls_to_resolve))
         for resolved_url in expected_urls_to_resolve:
             self.assertIn(resolved_url, urls_to_resolve)
+        if import_url == RETRY_URL:
+            self.assertEqual(DEFAULT_NUMBER_RETRIES, len(number_of_attempts))
```

Comment:
add comment stating the mock retry logic is expected to be 1 initial attempt + 4 retries